DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -296,23 +336,47 @@ function UninstallPackage
 
 function IsPackageInstalled
 {
+    [CmdletBinding(DefaultParameterSetName = 'RequiredVersion')]
+    [OutputType([Boolean])]
     param(
-        [Parameter(Position=0,Mandatory)][string]$pName,
-        [Parameter(Position=1)][string]$pVersion
+        [Parameter(Position=0, Mandatory)]
+        [string]$pName,
+        
+        [Parameter(ParameterSetName = 'RequiredVersion')]
+        [string]$pVersion,
+        
+        [Parameter(ParameterSetName = 'MinimumVersion')]
+        [string]$pMinimumVersion
     )
     Write-Verbose -Message "Start IsPackageInstalled $pName"
 
     $env:Path = [Environment]::GetEnvironmentVariable('Path','Machine')
-    Write-Verbose -Message "Path variables: $env:Path"
+    Write-Verbose -Message "Path variables: $($env:Path)"
 
     $installedPackages = Get-ChocoInstalledPackage
 
     if ($pVersion) {
-        Write-Verbose 'Comparing version'
-        $installedPackages = $installedPackages | Where-object { $_.Name -eq $pName -and $_.Version -eq $pVersion}
-    } else {
+        Write-Verbose 'Comparing required version'
+        $installedPackages = $installedPackages | Where-Object { $_.Name -eq $pName -and $_.Version -eq $pVersion}
+    }
+    elseif ($pMinimumVersion) {
+        Write-Verbose 'Comparing minimum version'
+        # version comparison can be done with [System.Version] but this lacks the ability to compare pre-release versions
+        # because of this limitation MinimumVersion cannot be used in conjuction with pre-release packages
+        $pre = ($pMinimumVersion -split "-")[1]
+        if ($pre) {
+            throw "MinimumVersion does not support comparing pre-releases, please use Version parameter instead"
+        }
+        $comparablePackages = $installedPackages | Where-Object { $_.Name -eq $pName} | ForEach-Object {
+            # ignoring pre-release as per above comment
+            $v = [System.Version](($_.Version -split "-")[0])

```

Comment:
Could you add some more comments around this as it took a little time for me to get my head around it - explain what you are doing and why?

Reply:
added, hope that is clearer now.